src/runtime/plugins/autoLitWrapper.ts

```diff
@@ -1,44 +1,77 @@
 import { pathToFileURL } from "node:url";
-import { parseQuery, parseURL } from "ufo";
-import MagicString from "magic-string";
+import { parseURL } from "ufo";
+import { parse, walk, ELEMENT_NODE, Node, render } from "ultrahtml";
+import { d } from "vue-bundle-renderer/dist/types-dfcc483f";
 import type { NuxtSsrLitOptions } from "../../module";
 
-interface AutoLitWrapperOptions extends NuxtSsrLitOptions {
-  sourcemap?: {
-    server: boolean;
-    client: boolean;
-  };
-}
+const V_FOR_ATTRIBUTE = "v-for";
+const V_FOR_KEY_ATTRIBUTE = ":key";
 
-export default function autoLitWrapper({
-  litElementPrefix = "",
-  sourcemap = { client: false, server: true }
-}: AutoLitWrapperOptions) {
+export default function autoLitWrapper({ litElementPrefix = [] }: NuxtSsrLitOptions) {
   return {
     name: "autoLitWrapper",
-    transform(code: string, id: string) {
-      const { pathname, search } = parseURL(decodeURIComponent(pathToFileURL(id).href));
-      const { type } = parseQuery(search);
+    async transform(code: string, id: string) {
+      const { pathname } = parseURL(decodeURIComponent(pathToFileURL(id).href));
+      const isVueFile = pathname.endsWith(".vue");
+
+      if (!isVueFile || id.includes("macro=true")) {
+        return;
+      }
+
+      const template = code.match(/<template>([\s\S]*)<\/template>/);
 
-      const isVueFile = pathname.endsWith(".vue") && (type === "script" || !search);
+      if (!template) {
+        return;
+      }
 
-      if (!isVueFile) {
+      if (!litElementPrefix.some((prefix) => code.includes(`<${prefix}`))) {
         return;
       }
 
-      const prefix = Array.isArray(litElementPrefix) ? `(${litElementPrefix.join("|")})` : litElementPrefix;
+      const prefixRegex = new RegExp(`^(${litElementPrefix.join("|")})`, "i");
+
+      const ast = parse(code);
+
+      await walk(ast, (node) => {
+        if (node.type !== ELEMENT_NODE || !prefixRegex.test(node.name)) {
+          return;
+        }
+
+        const attributes = node.attributes;
+
+        const hasVFor = node.attributes["v-for"];
+
+        const wrapper: Node = {
+          name: "LitWrapper",
+          type: ELEMENT_NODE,
+          parent: node.parent,
+          children: [node],
+          attributes: {},
+          loc: node.loc
+        };
+
+        if (hasVFor) {
+          wrapper.attributes[V_FOR_ATTRIBUTE] = attributes[V_FOR_ATTRIBUTE];
+          wrapper.attributes[V_FOR_KEY_ATTRIBUTE] = attributes[V_FOR_KEY_ATTRIBUTE];
+
+          delete node.attributes[V_FOR_ATTRIBUTE];
+          delete node.attributes[V_FOR_KEY_ATTRIBUTE];
+        }
+
+        Object.keys(node.attributes).forEach((key) => {
+          if (key === "") {
+            delete node.attributes[key];
+          }
+        });
```

Comment:
Not sure if this works, but it's simpler:
```suggestion
        delete node.attributes[""];
```

Reply:
Done!